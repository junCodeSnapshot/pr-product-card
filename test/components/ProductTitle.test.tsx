import React from "react";
import ReactTestRenderer from "react-test-renderer";
import { ProductTitle, ProductCart } from '../../src/components';
import { product1 } from "../data/productData";



describe('Test on ProductTitle', () => {

    test('should show the component right with the custom title', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductTitle title="Custom-title" className="customTestClassname"/>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('should show the component with the product name', () => {
        const wrapper = ReactTestRenderer.create(
            <ProductCart product={product1}>
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCart>
        );
        expect(wrapper.toJSON()).toMatchSnapshot();
    })

})
