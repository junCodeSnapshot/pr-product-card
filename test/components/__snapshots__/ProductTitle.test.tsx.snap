// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on ProductTitle should show the component right with the custom title 1`] = `
<span
  className="productDescription customTestClassname"
>
  Custom-title
</span>
`;

exports[`Test on ProductTitle should show the component with the product name 1`] = `
<div
  className=" productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - without image
  </span>
</div>
`;
